### PR TITLE
12199 Update Doc List: Suppress output for Coop Annual Report Correction for all Coop records

### PR DESCRIPTION
**Pull request by lewischenstudio**
*Issue #:* [/bcgov/entity#12199](https://app.zenhub.com/workspaces/entities-team-space-6143567664fb320019b81f39/issues/bcgov/entity/12199)

*Description of changes:*
- suppress `Notice of Articles` item for "Correction - Annual Report" in Recent Filing History, if 
  - the filing type is `Coop`,
  - the display name is `Correction - Annual Report`, and
  - the document link is for `noticeOfArticles`

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

**Review by severinbeauvais — CHANGES_REQUESTED** [review]
This is not the correct solution.

The list of outputs is provided by the Legal API. Hence, the Legal API should not include this output in this scenario.

File: src/components/Dashboard/FilingHistoryList.vue [hunk 1]
```diff
@@ -882,6 +882,15 @@ export default class FilingHistoryList extends Mixins(
     )
   }
 
+  /** Check whether to supprese notice of articles. */
+  private suppressNOA (displayName:string, prop: string): boolean {
+    // Only suppress NOA when it meets the following conditions
+    const type = this.getEntityType === CorpTypeCd.COOP
+    const name = displayName === 'Correction - Annual Report'
+    const isNOA = prop === 'noticeOfArticles'
```
Comment (severinbeauvais) [inline src/components/Dashboard/FilingHistoryList.vue:890]: If this was the right solution then you should use existing getters and enums as much as possible.

Comment (lewischenstudio) [reply src/components/Dashboard/FilingHistoryList.vue:890]: Yeah. I was looking for enums for the `displayName` and the document name, but we don't have them. They are however available in the backend.

**Comment by vysakh-menon-aot** [comment]
> This is not the correct solution.
> 
> The list of outputs is provided by the Legal API. Hence, the Legal API should not include this output in this scenario.

@lewischenstudio look into this code. https://github.com/bcgov/lear/blob/main/legal-api/src/legal_api/core/filing.py#L454
i think you just need to add an or condition to it.

**Comment by lewischenstudio** [comment]
> > This is not the correct solution.
> > The list of outputs is provided by the Legal API. Hence, the Legal API should not include this output in this scenario.
> 
> @lewischenstudio look into this code. https://github.com/bcgov/lear/blob/main/legal-api/src/legal_api/core/filing.py#L454 i think you just need to add an or condition to it since the ticket say's `Suppress all outputs`.

I verified this ticket with Mihai that we only need to suppress "Notice of Articles" and keep the rest of the outputs.

**Comment by lewischenstudio** [comment]
> This is not the correct solution.
> 
> The list of outputs is provided by the Legal API. Hence, the Legal API should not include this output in this scenario.

There are two places to work on this ticket, and I wasn't sure if I should use the FE or BE. I guess we can use the legal-api for this ticket then.

**Comment by lewischenstudio** [comment]
> > This is not the correct solution.
> > The list of outputs is provided by the Legal API. Hence, the Legal API should not include this output in this scenario.
> 
> @lewischenstudio look into this code. https://github.com/bcgov/lear/blob/main/legal-api/src/legal_api/core/filing.py#L454 i think you just need to add an or condition to it since the ticket say's `Suppress all outputs`.

For only to suppress NOA, I think I need to modify under this if condition (https://github.com/bcgov/lear/blob/e259eb1fb42c77b8e43b020081ef25558ff5ceea/legal-api/src/legal_api/core/meta/filing.py#L334-339), since it's for correction type.

**Comment by vysakh-menon-aot** [comment]
> > > This is not the correct solution.
> > > The list of outputs is provided by the Legal API. Hence, the Legal API should not include this output in this scenario.
> > 
> > 
> > @lewischenstudio look into this code. https://github.com/bcgov/lear/blob/main/legal-api/src/legal_api/core/filing.py#L454 i think you just need to add an or condition to it since the ticket say's `Suppress all outputs`.
> 
> For only to suppress NOA, I think I need to modify under this if condition [334-339](https://github.com/bcgov/lear/blob/e259eb1fb42c77b8e43b020081ef25558ff5ceea/legal-api/src/legal_api/core/meta/filing.py#L334-339), since it's for correction type.

Yes

**Pull request closed without merging**